addons/hr_expense/models/hr_expense.py

```diff
@@ -1134,12 +1134,21 @@ def _get_responsible_for_approval(self):
         return self.env['res.users']
 
     def activity_update(self):
-        for expense_report in self.filtered(lambda hol: hol.state == 'submit'):
-            self.activity_schedule(
-                'hr_expense.mail_act_expense_approval',
-                user_id=expense_report.sudo()._get_responsible_for_approval().id or self.env.user.id)
-        self.filtered(lambda hol: hol.state == 'approve').activity_feedback(['hr_expense.mail_act_expense_approval'])
-        self.filtered(lambda hol: hol.state in ('draft', 'cancel')).activity_unlink(['hr_expense.mail_act_expense_approval'])
+        reports_requiring_feedback = self.env['hr.expense.sheet']
+        reports_activity_unlink = self.env['hr.expense.sheet']
+        for expense_report in self:
+            if expense_report.state == 'submit':
+                self.activity_schedule(
```

Comment:
```suggestion
                expense_report.activity_schedule(
```
I believe the previous code was wrongly creating unwanted activities (on all self records) on multi recordset updates :thinking:.
(And the other places with an `activity_update` concept correctly use the unique record for the `activity_schedule` call)